### PR TITLE
[minizip-ng] Updated minizip version and fixed windows build for previous version

### PR DESCRIPTION
**Pull request by prateek9623**
**Describe the pull request**

- #### What does your PR fix?  
   Updated minizip version 
   Fixed windows build for the previous version( was only exporting symbols for compat header, [Suggested fix by author](https://github.com/zlib-ng/minizip-ng/issues/475))

- #### Which triplets are supported/not supported? Have you updated the [CI baseline]
   all except uwp, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

File: ports/minizip-ng/portfile.cmake [hunk 1]
```diff
@@ -28,6 +28,7 @@ vcpkg_cmake_configure(
         ${FEATURE_OPTIONS}
         -DMZ_FETCH_LIBS=OFF
         -DMZ_PROJECT_SUFFIX:STRING=-ng
+        -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=ON
```
Comment (JonLiu1993) [inline ports/minizip-ng/portfile.cmake:31]: CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS is a highly problematic behavior for many reasons, therefore our maintainer guide explicitly forbids adding it to libraries: https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#do-not-add-cmake_windows_export_all_symbols.

File: versions/m-/minizip-ng.json [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "d56c5b90261854568103947db593ed6e7342c3d2",
+      "version": "3.0.5",
+      "port-version": 0
+    },
+    {
+      "git-tree": "75746f0247b5654b873128c8dfcb5a5429cf758d",
+      "version": "3.0.2",
+      "port-version": 2
+    },
```
Comment (JonLiu1993) [inline versions/m-/minizip-ng.json:12]:
```suggestion
```

Comment (JonLiu1993) [inline versions/m-/minizip-ng.json:12]:
```suggestion
```

File: ports/minizip-ng/portfile.cmake (hunk 1 above)
Comment (prateek9623) [reply ports/minizip-ng/portfile.cmake:31]: @[nmoinvaz](https://github.com/nmoinvaz) How should we resolve this. Right now we can only static build for windows. I think we should expose only symbols using `__declspec`

File: versions/m-/minizip-ng.json [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "4b630863d08d630b64c928505a3fc9f9c7fe6dec",
+      "version": "3.0.5",
+      "port-version": 0
+    },
+    {
+      "git-tree": "a97e8cb9cc1a631a34e669b522291e21ec9d5ccd",
+      "version": "3.0.2",
+      "port-version": 2
+    },
```
Comment (JonLiu1993) [inline versions/m-/minizip-ng.json:12]:
```suggestion

```

File: ports/minizip-ng/portfile.cmake (hunk 1 above)
Comment (prateek9623) [reply ports/minizip-ng/portfile.cmake]: @JonLiu1993 It seems @[nmoinvaz](https://github.com/nmoinvaz) is not responding, for now, I have made changes to support static build only to resolve this issue.

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

File: ports/minizip-ng/portfile.cmake [hunk 4]
```diff
@@ -21,6 +21,10 @@ vcpkg_check_features(
         zstd MZ_ZSTD
 )
 
+if(WIN32)
+    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+endif()
+
```
Comment (JonLiu1993) [inline ports/minizip-ng/portfile.cmake:27]:
```suggestion
```

File: ports/minizip-ng/portfile.cmake [hunk 5]
```diff
@@ -1,8 +1,8 @@
 vcpkg_from_github(
```
Comment (JonLiu1993) [inline ports/minizip-ng/portfile.cmake:1]:
```suggestion
if (VCPKG_TARGET_IS_WINDOWS)
    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
endif()

vcpkg_from_github(
```

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**